### PR TITLE
Moving simple date logic to pinot

### PR DESCRIPTION
**Pull request by jenniferdai**

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java [hunk 1]
```diff
@@ -183,10 +183,24 @@ public void setTimeFormat(String timeFormat) {
     this._timeFormat = timeFormat;
   }
 
+  @Deprecated
   public String getTimeFormat() {
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java]: Nice. 
Please, also change the existing calls to getTimeFormat() to use the other calls u have introduced. Also, add a javadoc to getTimeFormat() pointing to the other methods to use.

Comment (jenniferdai) [reply pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java]: done, will also get a review from @npawar

File: pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java [hunk 2]
```diff
@@ -187,10 +188,38 @@ public void setTimeFormat(String timeFormat) {
...
+   * {@link #isEpochTimeFormat}
+   *
+   * To get Simple Date format, use {@link #getSimpleDateTimeFormat}
+   *
+   */
   public String getTimeFormat() {
     return _timeFormat;
   }
 
+  public boolean isSimpleDateFormat() {
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java]: Need @JsonIgnore here?

File: pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java [hunk 3]
```diff
@@ -187,10 +188,38 @@ public void setTimeFormat(String timeFormat) {
...
+    return _timeFormat.startsWith(TimeFormat.SIMPLE_DATE_FORMAT.toString());
+  }
+
+  public boolean isEpochTimeFormat() {
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java]: json ignore?

File: pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java [hunk 4]
```diff
@@ -187,10 +188,38 @@ public void setTimeFormat(String timeFormat) {
...
+    return _timeFormat.equals(TimeFormat.EPOCH.toString());
+  }
+
+  public SimpleDateFormat getSimpleDateTimeFormat() {
```
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java]: json ignore?

File: pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java [hunk 5]
```diff
@@ -187,10 +188,38 @@ public void setTimeFormat(String timeFormat) {
...
+    Preconditions.checkArgument(_timeFormat.startsWith(TimeFormat.SIMPLE_DATE_FORMAT.toString()),
+        "Not simple date format");
+    String simpleDateFormatStr = _timeFormat.split(":")[1];
```
Comment (npawar) [inline pinot-common/src/main/java/com/linkedin/pinot/common/data/TimeGranularitySpec.java]: Simple date format can have colons in it (eg: M/d/yyyy h:mm:ss). In such a case _timeFormat.split(":")[1] will extract only part of the format. Consider using split(separator, limit)?

**Pull request closed without merging**